lib/private/Files/ObjectStore/S3.php

```diff
@@ -69,13 +69,24 @@ public function uploadMultipartPart(string $urn, string $uploadId, int $partId,
 	}
 
 	public function getMultipartUploads(string $urn, string $uploadId): array {
-		$parts = $this->getConnection()->listParts([
-			'Bucket' => $this->bucket,
-			'Key' => $urn,
-			'UploadId' => $uploadId,
-			'MaxParts' => 10000
-		]);
-		return $parts->get('Parts') ?? [];
+		$parts = [];
+		$isTruncated = true;
+		$partNumberMarker = 0;
+
+		while ($isTruncated) {
+			$result = $this->getConnection()->listParts([
+				'Bucket' => $this->bucket,
+				'Key' => $urn,
+				'UploadId' => $uploadId,
+				'MaxParts' => 1000,
+				'PartNumberMarker' => $partNumberMarker
+			]);
+			$parts = array_merge($parts, $result->get('Parts'));
```

Comment:
Is it possible for get('Parts') to return null? Previous code had a coallescing operator.
```suggestion
			$parts = array_merge($parts, $result->get('Parts') ?? []);
```